okapi-core/src/main/java/org/folio/okapi/managers/ModuleManager.java

```diff
@@ -115,15 +115,31 @@ public ModuleManager(ModuleStore moduleStore, boolean local) {
           return Future.failedFuture(new OkapiError(ErrorType.USER, res));
         }
       }
-      return createList2(newList);
+      return createList2(newList).compose(x -> {
+        if (deleteObsolete) {
+          return deleteObsolete();
+        } else {
+          return Future.succeededFuture();
+        }
+      });
     });
   }
 
+  Future<Void> deleteObsolete() {
+    return modules.getAll()
+        .compose(ares -> {
+          List<ModuleDescriptor> newList = new LinkedList<>(ares.values());
```

Comment:
The LinkedList should be replaced by an ArrayList for faster sorting in getObsolete.